File: src/java/jnpp/service/dto/accounts/SavingBookDTO.java
package jnpp.service.dto.accounts;

import jnpp.dao.entities.accounts.SavingBookEntity;

public class SavingBookDTO {
    
    private final String name;
    private final Double moneyRate;
    private final Double timeRate;
    
    public SavingBookDTO(SavingBookEntity savingBook) {
        name = savingBook.getName();
        moneyRate = savingBook.getMoneyRate();
        timeRate = savingBook.getMoneyRate();
    }
    
    public String getName() {
        return name;
    }

    public Double getMoneyRate() {
        return moneyRate;
    }

    public Double getTimeRate() {
        return timeRate;
    }
    
}

File: src/main/webapp/commons/dto/accounts/saving-book-dto.js
class SavingBookDTO {

    constructor(json) {
        if (!json || !json.name || !json.moneyRate || !json.timeRate)
            throw new ParseException(json, typeof this);
        this.amount = json.amount;
        this.moneyRate = json.moneyRate;
        this.timeRate = json.timeRate;
    }

}
